compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6857,15 +6857,25 @@ private BType checkInvocationParam(BLangInvocation iExpr, AnalyzerData data) {
         int i = 0;
         BLangExpression vararg = null;
         boolean foundNamedArg = false;
+        boolean isIncRecordAllowExtraFields = incRecordParamAllowAdditionalFields != null;
         for (BLangExpression expr : iExpr.argExprs) {
             switch (expr.getKind()) {
                 case NAMED_ARGS_EXPR:
                     foundNamedArg = true;
-                    if (i < parameterCountForNamedArgs || incRecordParamAllowAdditionalFields != null) {
+                    BLangNamedArgsExpression namedArg = (BLangNamedArgsExpression) expr;
+                    boolean isNamedArgForIncRecordParam =
+                                  isNamedArgForIncRecordParam(namedArg.name.value, incRecordParamAllowAdditionalFields);
```

Comment:
For 

```ballerina
type Foo record {|
    int r;
|}; 

function fn(int r, *Foo x) {

}
```

we are getting `redeclared symbol 'r'`. While the error isn't the ideal error, shouldn't we get a similar error for

```ballerina
type Foo record {|
    int r;
|}; 

function fn(*Foo r) {

}
```

which may have resulted in special checks not being needed?

Reply:
Are you stating that this scenario is a compile-time error?
```ballerina
type Foo record {|
    int r;
|}; 

function fn(*Foo r) {

}
```

Reply:
If we are getting an error for

```ballerina
type Foo record {|
    int r;
|}; 

function fn(int r, *Foo x) {

}
```

shouldn't it be the same for

```ballerina
type Foo record {|
    int r;
|}; 

function fn(*Foo r) {

}
```
?

Why do we give an error for the first, btw? It makes sense, but I'm not sure if the spec mentions it.

Reply:
IIMO, giving an error for the first scenario is a bug because it can be invoked using position arguments. Therefore, if we try to invoke it with named arguments,error should be raised in that instance.
```ballerina
type Foo record {|
    int r;
|}; 

function fn(int r, *Foo x) {

}
public function main() {
    Foo f = {r: 1};
    fn(1, f);
}
```

Reply:
But then one can argue, `x` shouldn't be an included record parameter, right? Is this error specified in the spec? Assuming you were the one who added the error, do you recall why it was done that way?

Reply:
As  I remember, we consider fields of the record parameter as parameters of the function. I have made both scenarios into error cases.

Reply:
Is there anything in the spec (or spec repo issue) requiring these to be disallowed?

Reply:
The specification mentions that `the field names of all the parameter field descriptors whose type is not never must be distinct from each other and from the names of the parameters`

Reply:
This becomes a backward-incompatible change then, though. Please make a note to include it in the release note.